torch_geometric/data/temporal.py

```diff
@@ -111,17 +111,21 @@ def __prepare_non_str_idx(idx):
                 f'{type(idx).__name__}).')
         return idx
 
+    def __generate_item(self, idx):
+        data = {}
+        num_events = self.num_events
+        for key, item in self._store.items():
+            if item.size(0) == num_events:
+                data[key] = item[idx]
+        return TemporalData(**data)
```

Comment:
What's the reason we do not `copy` here anymore? I think this is a little bit cleaner, as we maintain the original class and any original features (that are not of size `num_events`) that way.

Reply:
As said [here](https://github.com/pyg-team/pytorch_geometric/pull/3985#discussion_r802218213), it was to improve a bit the performance (and also memory utilization) of the iteration and item retrieval. I can revert to the previous implementation if you think its benefits are more than the disadvantages.